src/oauth.test.ts

```diff
@@ -65,9 +78,504 @@ describe.sequential("OAuth", () => {
       "client_secret_basic",
     ]);
 
-    expect(metadata.scopes_supported).to;
-
     expect(Array.isArray(metadata.scopes_supported)).toBeTruthy();
+    expect(metadata.scopes_supported).toEqual(scopeEnum.enumValues);
+
+    expect(
+      Array.isArray(metadata.code_challenge_methods_supported),
+    ).toBeTruthy();
+    expect(metadata.code_challenge_methods_supported).toEqual(["S256"]);
+  });
+
+  describe.sequential("GET /oauth/authorize", () => {
+    let application: Schema.Application;
+    let client: Awaited<ReturnType<typeof createOAuthApplication>>;
+    let account: Awaited<ReturnType<typeof createAccount>>;
+
+    beforeEach(async () => {
+      account = await createAccount();
+      client = await createOAuthApplication({
+        scopes: ["read", "read:accounts", "follow"],
+        redirectUris: [OOB_REDIRECT_URI, "http://app.example/"],
+        confidential: true,
+      });
+      application = await getApplication(client);
+    });
+
+    it("successfully displays an authorization page", async () => {
+      expect.assertions(14);
+
+      const cookie = await getLoginCookie();
+      const parameters = new URLSearchParams();
+
+      parameters.set("response_type", "code");
+      parameters.set("client_id", application.clientId);
+      parameters.set("redirect_uri", OOB_REDIRECT_URI);
+      parameters.set("scope", "read:accounts follow");
+
+      const response = await app.request(
+        `/oauth/authorize?${parameters.toString()}`,
+        {
+          method: "GET",
+          headers: {
+            Cookie: cookie,
+          },
+        },
+      );
+
+      expect(response.status).toBe(200);
+      expect(response.headers.get("content-type")).toMatch(/^text\/html/);
+
+      const page = await getPage(response);
+
+      const listedScopes = Array.from(
+        page.querySelectorAll("#scopes > li > code"),
+      ).map((scopeElement) => scopeElement.textContent);
+
+      expect(listedScopes).toEqual(["read:accounts", "follow"]);
+
+      const form = page.querySelector("form[method='post']");
+      expect(form).not.toBeNull();
+
+      if (!form) {
+        throw new Error("Invariant error: form was not null but not found");
+      }
```

Comment:
This is just an invariant for typescript.. `expect(form).not.toBeNull()` doesn't make `form` in typescript to be present, hence this invariant check.